packages/terra-clinical-onset-picker/src/OnsetPicker.jsx

```diff
@@ -0,0 +1,362 @@
+      this.props.onsetDateInputOnChange(newDate);
+    }
+  }
+
+  /*
+   * Update onset date when month changes
+   */
+  changeMonth(event) {
+    const newDate = moment(this.state.onsetDate).month(event.target.value);
+
+    this.setState({ onsetDate: newDate });
+
+    if (this.props.onsetDateInputOnChange) {
+      this.props.onsetDateInputOnChange(newDate);
+    }
+  }
+
+  /*
+   * Update onset date when date changes
+   */
+  changeDate(event, date) {
+    const newDate = moment(date);
+
+    this.setState({ onsetDate: newDate });
+
+    if (this.props.onsetDateInputOnChange) {
+      this.props.onsetDateInputOnChange(newDate);
+    }
+  }
+
+  render() {
+    const {
+      birthdate,
+      granularity,
+      granularitySelectName,
+      granularitySelectOnChange,
+      precision,
+      precisionSet,
+      precisionSelectName,
+      precisionSelectOnChange,
+      onsetDate,
+      onsetDateInputName,
+      onsetDateInputOnChange,
+      ...customProps
+    } = this.props;
+
+    const intl = this.context.intl;
+
+    let granularitySelect = null;
+    if (this.state.precision !== 'UNKNOWN') {
+      granularitySelect = (<SelectField
+        options={[{ value: 'AGE', display: intl.formatMessage({ id: 'Terra.onsetPicker.granularityAge' }) },
+                  { value: 'YEAR', display: intl.formatMessage({ id: 'Terra.onsetPicker.granularityYear' }) },
+                  { value: 'MONTH', display: intl.formatMessage({ id: 'Terra.onsetPicker.granularityMonth' }) },
+                  { value: 'DATE', display: intl.formatMessage({ id: 'Terra.onsetPicker.granularityDate' }) }]}
+        name={this.props.granularitySelectName}
+        defaultValue={this.state.granularity}
+        onChange={this.changeGranularity}
+        isInline
+      />);
+    }
+
+    let ageSelect = null;
+    let ageUnitSelect = null;
+    if (this.state.granularity === 'AGE') {
+      ageSelect = (<NumberField
+        data-terra-clinical-onset-picker="age"
+        min={0}
+        max={OnsetUtil.allowedAge(this.props.birthdate, this.state.ageUnit)}
+        step={1}
+        value={this.state.age}
+        onChange={this.changeAge}
+        isInline
+      />);
+
+      ageUnitSelect = (<SelectField
+        data-terra-clinical-onset-picker="age_unit"
+        options={OnsetUtil.allowedAgeUnits(this.props.birthdate)}
+        defaultValue={this.state.ageUnit.toString()}
+        onChange={this.changeAgeUnit}
+        isInline
+      />);
+    }
+
+    let monthSelect = null;
+    if (this.state.granularity === 'MONTH') {
+      monthSelect = (<SelectField
+        data-terra-clinical-onset-picker="month"
+        options={OnsetUtil.allowedMonths(intl, this.props.birthdate, this.state.onsetDate)}
+        defaultValue={moment(this.state.onsetDate).month().toString()}
+        onChange={this.changeMonth}
+        isInline
+      />);
+    }
+
+    let yearSelect = null;
+    if (this.state.granularity === 'YEAR' || this.state.granularity === 'MONTH') {
```

Comment:
Would we ever want to provide a number input instead of a dropdown option? It seems this would be a lot of work on the consumer to provide each desired year if it was more than 10+ year options.

Reply:
I wish I had written this down somewhere but I'm pretty sure UX wanted this as a dropdown.